Sources/NIO/EventLoop.swift

```diff
@@ -417,27 +417,34 @@ internal final class SelectableEventLoop: EventLoop {
         }
     }
 
-    /// Handle the given `IOEvent` for the `SelectableChannel`.
-    private func handleEvent<C: SelectableChannel>(_ ev: IOEvent, channel: C) {
+    /// Handle the given `SelectorEventSet` for the `SelectableChannel`.
+    private func handleEvent<C: SelectableChannel>(_ ev: SelectorEventSet, channel: C) {
         guard channel.selectable.isOpen else {
             return
         }
 
-        switch ev {
-        case .write:
-            channel.writable()
-        case .read:
-            channel.readable()
-        case .all:
+        if ev.contains(.write) {
             channel.writable()
 
             guard channel.selectable.isOpen else {
                 return
             }
+        }
+
+        if ev.contains(.read) {
             channel.readable()
-        case .none:
-            // spurious wakeup
-            break
+
+            guard channel.selectable.isOpen else {
+                return
+            }
+        }
+
+        if ev.contains(.readEOF) {
+            channel.readEOF()
+        }
+
+        if ev.contains(.reset) {
+            channel.reset()
```

Comment:
Given that `reset` will inevitably cause the writes to fail, is it better to have `reset` processed *first*? That ensures faster channel teardown, rather than spinning all the write machinery just to get a write failure.

Reply:
+1

Reply:
sounds sensible!